src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/Watchdog.cs

```diff
@@ -46,88 +38,76 @@ protected Watchdog()
 
         internal string Name => GetType().Name;
 
-        internal string IsEnabledId => $"{Name}.IsEnabled";
-
         internal string PeriodSecId => $"{Name}.PeriodSec";
 
-        protected async Task InitializeAsync(bool isEnabled, int periodSec, CancellationToken cancellationToken)
-        {
-            // wait till we can truly init
-            while (!_storageReady || _schemaInformation.Current < _minVersion)
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-                await Task.Delay(TimeSpan.FromSeconds(5), cancellationToken);
-            }
+        internal string LeasePeriodSecId => $"{Name}.LeasePeriodSec";
 
-            await InitParamsAsync(isEnabled, periodSec);
-        }
+        internal bool IsLeaseHolder => _watchdogLease.IsLeaseHolder;
+
+        internal string LeaseWorker => _watchdogLease.Worker;
 
-        public async Task ExecutePeriodicLoopAsync(CancellationToken cancellationToken)
+        internal double LeasePeriodSec => _watchdogLease.PeriodSec;
+
+        protected internal async Task StartAsync(bool allowRebalance, double periodSec, double leasePeriodSec, CancellationToken cancellationToken)
```

Comment:
Other frameworks such as [Functions TimerTrigger](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=python-v2%2Cin-process&pivots=programming-language-csharp) and [Hangfire](https://www.hangfire.io/) have low dependency and code overhead for the actual job implementation. Both also support running jobs via Timespans or CRON syntax and have reliable mechanisms to ensure execution at the defined periodic schedule. As we continue to add jobs, how is our current framework transferable to extendable to these design principles?